include/mfmg/dealii/anasazi.templates.hpp

```diff
@@ -0,0 +1,93 @@
+/*************************************************************************
+ * Copyright (c) 2017-2019 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef MFMG_ANASAZI_TEMPLATES_HPP
+#define MFMG_ANASAZI_TEMPLATES_HPP
+
+#include <mfmg/common/exceptions.hpp>
+
+#include "anasazi.hpp"
+#include "anasazi_traits.hpp"
+#include "belos_traits.hpp"
+#include <AnasaziBasicEigenproblem.hpp>
+#include <AnasaziFactory.hpp>
+
+namespace mfmg
+{
+
+/// \brief Anasazi solver: constructor
+template <typename OperatorType, typename VectorType>
+AnasaziSolver<OperatorType, VectorType>::AnasaziSolver(OperatorType const &op)
+    : _op(op)
+{
+  ASSERT(_op.m() == _op.n(), "Operator must be square");
+}
+
+/// \brief Anasazi solver: perform Anasazi solve, use random initial guess
+template <typename OperatorType, typename VectorType>
+std::tuple<std::vector<double>, std::vector<VectorType>>
+AnasaziSolver<OperatorType, VectorType>::solve(
+    boost::property_tree::ptree const &params, VectorType initial_guess) const
```

Comment:
Do you mean to pass it by value?

Reply:
Not sure, this was taken from the Lanczos solver.